.github/workflows/pr-issue-validator.yaml

```diff
@@ -31,128 +25,79 @@ jobs:
+          "((Fixes|Resolves) devtron-labs/devtron#[0-9]+)"
+          "(Fixes|Resolves):?\\s+\\[#([0-9]+)\\]" 
+          "((Fixes|Resolves):? #devtron-labs/devops-sprint/issues/[0-9]+)"
+          "((Fixes|Resolves):? #devtron-labs/sprint-tasks/issues/[0-9]+)" 
+          "((Fixes|Resolves) https://github.com/devtron-labs/devops-sprint/issues/[0-9]+)"
+          "((Fixes|Resolves) https://github.com/devtron-labs/sprint-tasks/issues/[0-9]+)"
+        )
+    
+        # Extract issue number and repo from PR body
+        extract_issue_number() {
+          local pattern="$1"
+          if [[ "$PR_BODY" =~ $pattern ]]; then
+            echo "matched for this pattern $pattern"
+            issue_num=$(echo "${BASH_REMATCH[0]}" | grep -oE "[0-9]+" | tr -d '\r\n')
+            repo=$(echo "$PR_BODY" | grep -oE "devtron-labs/[a-zA-Z0-9_-]+")
+            echo "Extracted issue number: $issue_num from repo: $repo"
+            return 0
+          else
+            echo "No match for the pattern $pattern"
+          fi
+          return 1
+        }
+    
+        issue_num=""
+        repo="devtron-labs/devtron"  # Default repo
+        for pattern in "${patterns[@]}"; do
+          echo "Now checking for $pattern"
+          extract_issue_number "$pattern" && break
+        done
+    
+        if [[ -z "$issue_num" ]]; then
+          echo "No valid issue number found."
+          gh pr edit $PRNUM --add-label "PR:Issue-verification-failed"
+          gh pr edit $PRNUM --remove-label "PR:Ready-to-Review"
+          exit 1
         fi
-
-        if echo "$PR_BODY" | grep -iEq "$pattern5|$pattern6|$pattern7|$pattern8"; then
-            echo $url
-        else
-            ### Here we are setting the Internal Field Separator to "/"
-            ### read -r -> reads input from variable $url
-            ### -a url_parts -> tells read command to store input into an array named url_parts[]
-            IFS="/" read -r -a url_parts <<< "$url"
-            
-            # Remove the last two elements (repos and the issue number)
-            unset url_parts[-1]
-            unset url_parts[-1]
-            # Reattach the URL pieces
-            url=$(IFS=/; echo "${url_parts[*]}")
-
-            # Add the issue number to the URL
-            url="${url}/issues/${issue_num}"
-            echo "$url"
-        fi
-        
-        response_code=$(curl -s -o /dev/null -w "%{http_code}" "$url")
+    
+        # Form the issue API URL dynamically
+        issue_api_url="https://api.github.com/repos/$repo/issues/$issue_num"
+        echo "API URL: $issue_api_url"
+    
+        # Check if the issue exists in the private repo
+        response_code=$(curl -s -o /dev/null -w "%{http_code}" \
+          --header "authorization: Bearer ${{ secrets.GH_PR_VALIDATOR_TOKEN }}" \
+          "$issue_api_url")
+          
         if [[ "$response_code" -eq 200 ]]; then
-          # Check if issue is open or closed
```

Comment:
Lets add back the logic to check if issue is open or close because we want to fail PR validation if we link closed issues.

Reply:
added